Source/WebKit/NetworkProcess/Notifications/NetworkNotificationManager.cpp

```diff
@@ -43,21 +43,11 @@ using namespace WebCore;
 
 WTF_MAKE_TZONE_ALLOCATED_IMPL(NetworkNotificationManager);
 
-NetworkNotificationManager::NetworkNotificationManager(NetworkSession& networkSession, const String& webPushMachServiceName, WebPushD::WebPushDaemonConnectionConfiguration&& configuration)
-    : m_networkSession(networkSession)
+NetworkNotificationManager::NetworkNotificationManager(const String& webPushMachServiceName, bool isEphemeralSession, WebPushD::WebPushDaemonConnectionConfiguration&& configuration)
+    : m_isEphemeralSession(isEphemeralSession)
 {
-    if (!m_networkSession.sessionID().isEphemeral() && !webPushMachServiceName.isEmpty()) {
-#if PLATFORM(COCOA)
-        auto token = m_networkSession.networkProcess().parentProcessConnection()->getAuditToken();
-        if (token) {
-            Vector<uint8_t> auditTokenData(sizeof(*token));
-            memcpy(auditTokenData.data(), &(*token), sizeof(*token));
-            configuration.hostAppAuditTokenData = WTFMove(auditTokenData);
-        }
-#endif
-
-        m_connection = makeUnique<WebPushD::Connection>(webPushMachServiceName.utf8(), *this, WTFMove(configuration));
-    }
+    if (!webPushMachServiceName.isEmpty() && !isEphemeralSession)
```

Comment:
Am I reading it right that if the session is ephemeral then `NetworkNotificationManager` is numb and does not do anything?
If so, should we just not create `NetworkNotificationManager` if session is ephemeral instead of passing `isEphemeralSession` to `NetworkNotificationManager`?